src/flows/CostCalculator/CostCalculator.tsx

```diff
@@ -0,0 +1,218 @@
+import { HeadlessFormOutput } from '@remoteoss/json-schema-form';
+import React, { useMemo, useRef } from 'react';
+import { useForm } from 'react-hook-form';
+import type { InferType } from 'yup';
+import { object, string } from 'yup';
+
+import type {
+  CostCalculatorEstimateParams,
+  CostCalculatorEstimateResponse,
+  EmploymentTermType,
+} from '@/src/client';
+import { Form } from '@/src/components/ui/form';
+
+import { SelectField } from '@/src/components/form/fields/SelectField';
+import { TextField } from '@/src/components/form/fields/TextField';
+import { JSONSchemaFormFields } from '@/src/components/form/JSONSchemaForm';
+import { useValidationFormResolver } from '@/src/components/form/yupValidationResolver';
+import { Button } from '@/src/components/ui/button';
+import {
+  useCalculatorLoadRegionFieldsSchemaForm,
+  useCompanyCurrencies,
+  useCostCalculatorCountries,
+  useCostCalculatorEstimation,
+} from '@/src/flows/CostCalculator/hooks';
+
+const validationSchema = object({
+  country: string().required('Country is required'),
+  currency: string().required('Currency is required'),
+  region: string(),
+  salary: string()
+    .typeError('Salary must be a number')
+    .required('Salary is required'),
+});
+
+type FormValues = InferType<typeof validationSchema> & {
+  contract_duration_type?: EmploymentTermType;
+  age?: number;
+};
+
+type Props = Partial<{
+  /**
+   * Estimation params allows you to customize the parameters sent to the /cost-calculator/estimation endpoint.
+   */
+  estimationParams: {
+    /**
+     * Title of the estimation. Default is 'My first estimation'.
+     */
+    title?: string;
+    /**
+     * Include benefits in the estimation. Default is false.
+     */
+    includeBenefits?: boolean;
+    /**
+     * Include cost breakdowns in the estimation. Default is false.
+     */
+    includeCostBreakdowns?: boolean;
+  };
+  /**
+   * Default values for the form fields.
+   */
+  defaultValues: {
+    /**
+     * Default value for the country field.
+     */
+    country?: string;
+    /**
+     * Default value for the currency field.
+     */
+    currency?: string;
+    /**
+     * Default value for the salary field.
+     */
+    salary?: string;
+  };
+  /**
+   * Callback function to handle the form submission. When the submit button is clicked, the payload is sent back to you.
+   * @param data - The payload sent to the /cost-calculator/estimation endpoint.
+   */
+  onSubmit: (data: CostCalculatorEstimateParams) => void;
+  /**
+   * Callback function to handle the success when the estimation succeeds. The CostCalculatorEstimateResponse is sent back to you.
+   * @param data - The response data from the /cost-calculator/estimation endpoint.
+   */
+  onSuccess: (data: CostCalculatorEstimateResponse) => void;
+  /**
+   * Callback function to handle the error when the estimation fails.
+   * @param error - The error object.
+   */
+  onError: (error: Error) => void;
```

Comment:
All callbacks should be optional 🙏 

```suggestion
  onSubmit?: (data: CostCalculatorEstimateParams) => void;
  /**
   * Callback function to handle the success when the estimation succeeds. The CostCalculatorEstimateResponse is sent back to you.
   * @param data - The response data from the /cost-calculator/estimation endpoint.
   */
  onSuccess?: (data: CostCalculatorEstimateResponse) => void;
  /**
   * Callback function to handle the error when the estimation fails.
   * @param error - The error object.
   */
  onError?: (error: Error) => void;
```

Reply:
doesn't the partial already do that?

Reply:
Ah right! Yes it does 🤦